pyflp/plugin.py

```diff
@@ -155,6 +156,49 @@ class FruitySendEvent(StructEventBase):
     ).compile()
 
 
+class FruitySlicerEvent(StructEventBase):
+    STRUCT = c.Struct(
+        "_u1" / c.Bytes(8),
+        "bpm" / c.Float32l,
+        "pitch_shift" / c.Int32sl,
+        "time_stretch" / c.Int32sl,
+        "stretching_method"
+        / c.Enum(
+            c.Int32ul,
+            fill_gaps=0,
+            alt_fill_gaps=1,
+            pro_default=2,
+            pro_transient=3,
+            transient=4,
+            tonal=5,
+            monophonic=6,
+            speech=7,
+        ),
+        "fade_in" / c.Int32ul,
+        "fade_out" / c.Int32ul,
+        "file_path" / c.PascalString(c.Int8ul, "utf-8"),
+        "slices"
+        / c.PrefixedArray(
+            c.Int32ul,
+            c.Struct(
+                "name" / c.PascalString(c.Int8ul, "utf-8"),
```

Comment:
Just checked these, the file path does seem to be a u8, but interestingly, the slice name field seems to be a bit unusual. The name is capped at 257 bytes, but it uses a different size format that seems to adapt to the actual length of the string. From what I can tell, the first byte is either the length of the string or 0xFF to indicate that the length is actually defined by the next four bytes (a u32). For example, hex `40` would indicate a string of 64 bytes, and the size field would only take up one byte. However, hex `ff01010000` would indicate a string of 257 bytes, and the size field would take up five bytes, the first simply defining the next four as the size. I haven't checked, and it doesn't really matter for this purpose, but it might be possible that this could extend to a u64 as well using the same pattern. Perhaps an adapter (`AdaptiveInteger`?) could be implemented for this?

Reply:
The file path does seem to be limited to 255 characters, but it gets weirder -- attempting to load a sample with a longer path results in FL Studio itself crashing with the memory error `free(): invalid next size (normal)`. I tested this in both FL 20 and 21 with the same result. It's probably safe to say that the path is a u8, but if you can't reproduce this, I'm not sure what's going on here.

Reply:
I do have enabled, but I believe the reason it was crashing may have been due to Wine (I run Linux). With a unique file path of 259 characters, it seemed to use the same format as the slice names (the data field was FF 03 01 00 00). I assume maybe it shortens paths that are 260 characters or longer, but keeps paths that are longer than 260 characters the same, where this obscure format is only used for paths between 255 and 259 characters.